make: Move CPU and BOARD includes to Makefile.modules and reorganize

gebart opened the pull request:
Removed boards/Makefile.cortexm_common in favour of cpu/Makefile.cortexm_common
I always found it confusing to look for my CPU compiler settings in the boards Makefile.

buildtested on my machine, binary runs on a mulle board.

haukepetersen commented on cpu/Makefile.include.cortexm_common:
```diff
@@ -70,6 +120,14 @@ endif
 include $(RIOTCPU)/cortexm_common/Makefile.include
 
+# Import all toolchain settings
+include $(RIOTBOARD)/Makefile.include.$(TOOLCHAIN)
```
to be consistent, we should move the toolchain settings Makefile also into the CPU folder

gebart replied:
I agree, although I would like to take it one step further and move them all into a separate makefile directory, see #4699

haukepetersen commented:
+1 for this change (although if I remember correctly, there was a reason for having the toolchain configuration tied to the board. @OlegHahm do you remember why?)

two things that come to mind:
1. shouldn't we move `$(RIOTBOARD)/$(BOARD)/include` and `$(RIOTCPU)/$(CPU)/include` to `Makefile.modules`? This way we could remove the adding of them from many many cpu and board makefiles
2. I think this change should in the same PR also be done for the other platforms (e.g. msp430, arm7, etc)

OlegHahm commented:
I cannot remember right now.

kaspar030 commented on boards/airfy-beacon/Makefile.include:
```diff
@@ -12,5 +12,5 @@ include $(RIOTBOARD)/Makefile.include.serial
 
-# include cortex defaults
-include $(RIOTBOARD)/Makefile.include.cortexm_common
+# export board specific includes to the global includes-listing
```
As these two lines are identical to all boards, can't they go into the cpu Makefile?

gebart commented:
@haukepetersen 
 - Rebased
 - moved the CPU and BOARD includes to Makefile.modules
 - moved Makefile.include.msp430_common to cpu dir
 - moved Makefile.include.$(TOOLCHAIN) to cpu dir
 - (Updated title)

haukepetersen commented:
Nice, looking great and the build output from some selected boards looks as expected. ACK once Travis is ok with it.

gebart commented:
I was getting random Travis failures on the nrf51 platforms, rebased in hope of fixing it. Building is working locally for me.

haukepetersen commented:
We are getting these Nordic fails everywhere, but I have no idea what Travis is doing. Locally everything Travis complains about works fine, so no idea whats causing this

gebart commented:
I added some debugging to the buildtest target in an attempt to find the cause

gebart commented:
but of course it doesn't randomly fail after I add debugging output... OK to merge after I remove the debugging crap again?

OlegHahm commented:
Is it something like
```
malloc:airfy-beacon:

Building application "test_malloc" for "airfy-beacon" with MCU "nrf51".

/home/travis/build/RIOT-OS/RIOT/tests/malloc/bin/airfy-beacon/uart_stdio.a(uart_stdio.o): In function `uart_stdio_init':

/home/travis/build/RIOT-OS/RIOT/sys/uart_stdio/uart_stdio.c:76: undefined reference to `uart_init'

/home/travis/build/RIOT-OS/RIOT/tests/malloc/bin/airfy-beacon/uart_stdio.a(uart_stdio.o): In function `uart_stdio_write':

/home/travis/build/RIOT-OS/RIOT/sys/uart_stdio/uart_stdio.c:94: undefined reference to `uart_write'

collect2: error: ld returned 1 exit status

make: *** [all] Error 1
```
you're talking about? Just found this in another PR, here https://travis-ci.org/RIOT-OS/RIOT/jobs/108422976

gebart commented:
yes, I did the following change to Makefile.buildtests to better see what happens:

https://github.com/gebart/RIOT/commit/530704b451745e2f92afd8de77737ec9c0f38958

gebart commented:
After removing the debug commit and restarting Travis all seem fine, I have not made any changes to the commit lineup since @haukepetersen's ACK so I'll merge this now.
